test/fixtures/poro.rb

```diff
@@ -30,67 +31,74 @@ def respond_to_missing?(method_name, _include_private = false)
 #   model = ModelWithErrors.new
 #   model.validate!            # => ["cannot be nil"]
 #   model.errors.full_messages # => ["name cannot be nil"]
-class ModelWithErrors < ::ActiveModelSerializers::Model
+class ModelWithErrors < Model
   attributes :name
 end
 
 class Profile < Model
+  attributes :name, :description, :comments
 end
-
 class ProfileSerializer < ActiveModel::Serializer
   attributes :name, :description
 
-  # TODO: is this used anywhere?
+  # ActiveModel::Serializer::OptionsTest#test_options_are_accessible:
+  # test/serializers/options_test.rb:12:in `test_options_are_accessible'
+  # test/serializers/options_test.rb:17:in `test_no_option_is_passed_in'
   def arguments_passed_in?
     instance_options[:my_options] == :accessible
   end
 end
-
 class ProfilePreviewSerializer < ActiveModel::Serializer
   attributes :name
 end
 
-class Post < Model; end
-class Like < Model; end
-class Author < Model; end
-class Bio < Model; end
-class Blog < Model; end
-class Role < Model; end
-class User < Model; end
-class Location < Model; end
-class Place < Model; end
-class Tag < Model; end
-class VirtualValue < Model; end
+class Author < Model
+  attributes :name
+  associations :posts, :bio, :roles, :comments
+end
+class AuthorSerializer < ActiveModel::Serializer
+  cache key: 'writer', skip_digest: true
+  attribute :id
+  attribute :name
+
+  has_many :posts
+  has_many :roles
+  has_one :bio
+end
+class AuthorPreviewSerializer < ActiveModel::Serializer
+  attributes :id
+  has_many :posts
+end
+
 class Comment < Model
+  attributes :body, :date
+  associations :post, :author, :likes
+
   # Uses a custom non-time-based cache key
   def cache_key
     "#{self.class.name.downcase}/#{id}"
   end
 end
+class CommentSerializer < ActiveModel::Serializer
+  cache expires_in: 1.day, skip_digest: true
+  attributes :id, :body
+  belongs_to :post
+  belongs_to :author
 
-class Employee < ActiveRecord::Base
-  has_many :pictures, as: :imageable
-  has_many :object_tags, as: :taggable
-end
-
-class ObjectTag < ActiveRecord::Base
-  belongs_to :poly_tag
-  belongs_to :taggable, polymorphic: true
-end
-
-class Picture < ActiveRecord::Base
-  belongs_to :imageable, polymorphic: true
-  has_many :object_tags, as: :taggable
+  def custom_options
```

Comment:
I guess it is used to test that options are passed from collection serializer -> serializer and serializer -> association_serializer but it could definitely benefit from creating a specific serializer within the file testing this case.